### ISSUE TITLE
syntax highlighting error:  concatenated strings

### ISSUE THREAD

**Issue by guwirth** [SonarOpenCommunity/sonar-cxx#1094]
Get a syntax highlighting error with the sample below:

```C++
#define T_STRAT_BRAC        "["
#define T_END_BRAC          "]"
#define T_ST_NOT_SUP    T_STRAT_BRAC"START_NOT_SUPPORTED"T_END_BRAC

void test()
{
    fileOutput << T_ST_NOT_SUP << endl;
}
```
The sonar-scanner analysis succeeds but server logs from background tasks mention:
```
[o.s.s.c.s.HighlightingLineReader] Inconsistency detected in Highlighting data. Highlighting will be ignored for file 'com.org.XXX:XXX:OSS/Sources/sample.cpp'
org.sonar.server.computation.source.RangeOffsetConverter$RangeOffsetConverterException: End offset 40 is defined outside the length (39) of the line 1392
        at org.sonar.server.computation.source.RangeOffsetConverter.checkExpression(RangeOffsetConverter.java:65) ~[sonar-server-5.6.4.jar:na]
        at org.sonar.server.computation.source.RangeOffsetConverter.validateEndOffsetNotGreaterThanLineLength(RangeOffsetConverter.java:59) ~[sonar-server-5.6.4.jar:na]
        at org.sonar.server.computation.source.RangeOffsetConverter.offsetToString(RangeOffsetConverter.java:34) ~[sonar-server-5.6.4.jar:na]
        at org.sonar.server.computation.source.HighlightingLineReader.processHighlighting(HighlightingLineReader.java:104) ~[sonar-server-5.6.4.jar:na]
        at org.sonar.server.computation.source.HighlightingLineReader.processHighlightings(HighlightingLineReader.java:91) ~[sonar-server-5.6.4.jar:na]
        at org.sonar.server.computation.source.HighlightingLineReader.read(HighlightingLineReader.java:78) ~[sonar-server-5.6.4.jar:na]
        at org.sonar.server.computation.source.ComputeFileSourceData.read(ComputeFileSourceData.java:61) [sonar-server-5.6.4.jar:na]
```

**Comment by guwirth** [comment]
My first feeling is that `isGeneratedCode` flag is missing?

**Comment by jmecosta** [comment]
what flag is that isGeneratedCode, just curious as i have no idea what going on there.

**Comment by guwirth** [comment]
@jmecosta 

Sample in preprocessor:
https://github.com/SonarOpenCommunity/sonar-cxx/blob/master/cxx-squid/src/main/java/org/sonar/cxx/preprocessor/CxxPreprocessor.java#L1006

Generated code can't be highlighted:
https://github.com/SonarOpenCommunity/sonar-cxx/blob/056fbb3fcad753e658d0cc0df6603edf63b25d21/sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/highlighter/CxxHighlighter.java#L140

For concatenated strings there seems to be something wrong?

**Comment by guwirth** [comment]
closed with #1163
